src/main/java/com/example/decafe/HelloController.java

```diff
@@ -0,0 +1,577 @@
+                    }
+                    customer.getCoinImage().setImage(createImage(moneyImage)); //set coin image
+                    customer.getCoinImage().setOnMouseClicked(event1 -> { // set click event for coin image
+                        try {
+                            getMoney(event1, customer); // if coin Image is clicked jump to this method
+                        } catch (IOException e) {
+                            try {
+                                switchToEndScreen();
+                            } catch (IOException ex) {
+                                ex.printStackTrace();
+                            }
+                        }
+                    });
+                }
+            }
+        }
+    }
+
+    // Method to check if an Upgrade can be made (check if player has earned enough coins and if it was already used or not)
+    public void checkUpgradePossible(Upgrade upgrade) throws FileNotFoundException {
+        Play.checkUpgradePossible(upgrade);
+    }
+
+    // Method to actively use an Upgrade
+    public void doUpgrade(MouseEvent e) throws FileNotFoundException {
+        // activate the Upgrade (according to whatever ImageView was chosen)
+        Play.doUpgrade(((ImageView) e.getSource()).getId(), CofiBrew);
+        // set the coin label to the correct amount of coins (coins earned - upgrade costs)
+        coinsEarnedLabel.setText(String.valueOf(Play.getCoinsEarned()));
+
+        File f = new File("");
+        String musicFile = f.getAbsolutePath() + File.separator + "src" + File.separator + "main" + File.separator + "resources" + File.separator + "com" + File.separator + "example" + File.separator + "decafe" + File.separator + "upgradeSound.wav";
+        AudioClip getUpgrade = new AudioClip(new File(musicFile).toURI().toString());
+        //MediaPlayer collectMoney = new MediaPlayer(sound);
+        getUpgrade.play();
+
+        // check if other upgrades are still possible or if they need to be "deactivated"
+        checkUpgradePossible(Play.getCoffeeUpgrade());
+        checkUpgradePossible(Play.getCakeUpgrade());
+        checkUpgradePossible(Play.getPlayerUpgrade());
+    }
+
+    // check if collisions occur
+    public boolean checkForCollision(ImageView waiter) {
+        for (Label collision : collisions) { // iterate through labels
+            if (waiter.getBoundsInParent().intersects(collision.getBoundsInParent())) { // if waiter would collide with a label
+                return true; // return true
+            }
+        }
+        return false;
+    }
+
+    // Method used when coin Image is clicked on
+    public void getMoney(MouseEvent e, Customer customer) throws IOException {
+        File f = new File("");
+        String musicFile = f.getAbsolutePath() + File.separator + "src" + File.separator + "main" + File.separator + "resources" + File.separator + "com" + File.separator + "example" + File.separator + "decafe" + File.separator + "coinsSound.wav";
```

Comment:
there is another String with the variable name musicFile at line 137, therefore renaming would avoid confusion....